### PR TITLE
New message types for ipython-kernel

### PR DESCRIPTION
**Pull request by thomasjm**
Background: I've been working on something that's essentially a new Jupyter frontend in Haskell, and I needed several more IPython message types to be supported in `ipython-kernel`. This is kind of an ongoing process, but I thought I'd get a PR out with the current changes.

### CODE REVIEW

**Comment by gibiansky** [comment]
This looks good, but

1. Why is `ghc-parser` in `.gitignore`? It's a package we want in our repository.
2. There should not be more than one `stack.yaml`. There's already one for `ihaskell`, we don't need one for `ipython-kernel`, just use the same one.

**Comment by thomasjm** [comment]
1) the ghc-parser was getting randomly updated by the build process (I was finding strings with my home directory in it). Removed from .gitignore.

2) I thought I needed it to use `ipython-kernel` as a local stack package. Removed now.

File: ihaskell-display/ihaskell-diagrams/IHaskell/Display/Diagrams.hs [hunk 1]
```diff
@@ -29,12 +30,17 @@ diagramData renderable format = do
...
+
   -- Write the image.
   let filename = ".ihaskell-diagram." ++ extension format
   renderCairo filename (mkSizeSpec2D (Just imgWidth) (Just imgHeight)) renderable
 
+  putStrLn "Did renderCairo, about to readFile"
+
   -- Convert to base64.
   imgData <- Char.readFile filename
+  putStrLn "Did readFile"
```
Comment (gibiansky) [inline ihaskell-display/ihaskell-diagrams/IHaskell/Display/Diagrams.hs]: Debugging comments :)

File: ipython-kernel/src/IHaskell/IPython/Message/Parser.hs [hunk 2]
```diff
@@ -7,13 +7,16 @@
 -- the low-level 0MQ interface.
 module IHaskell.IPython.Message.Parser (parseMessage) where
 
-import           Data.Aeson ((.:), decode, Result(..), Object)
-import           Control.Applicative ((<|>), (<$>), (<*>))
-import           Data.Aeson.Types (parse)
-import           Data.ByteString
-import           Data.Map (Map)
-import           Data.Text (Text)
+-- import           Control.Applicative ((<|>), (<$>), (<*>))
```
Comment (gibiansky) [inline ipython-kernel/src/IHaskell/IPython/Message/Parser.hs]: uncomment or remove

File: ipython-kernel/src/IHaskell/IPython/Message/UUID.hs [hunk 3]
```diff
@@ -4,7 +4,7 @@
 module IHaskell.IPython.Message.UUID (UUID, random, randoms, uuidToString) where
 
 import           Control.Monad (mzero, replicateM)
-import           Control.Applicative ((<$>))
+-- import           Control.Applicative ((<$>))
```
Comment (gibiansky) [inline ipython-kernel/src/IHaskell/IPython/Message/UUID.hs]: uncomment or remove

File: ipython-kernel/src/IHaskell/IPython/Message/Writer.hs [hunk 4]
```diff
@@ -7,12 +7,12 @@ module IHaskell.IPython.Message.Writer (ToJSON(..)) where
...
 import           Data.Aeson
 import           Data.Map (Map)
+-- import           Data.Monoid (mempty)
 import           Data.Text (Text, pack)
-import           Data.Monoid (mempty)
 
-import qualified Data.ByteString.Lazy as L
-import qualified Data.ByteString as B
-import           Data.Text.Encoding
+-- import qualified Data.ByteString as B
```
Comment (gibiansky) [inline ipython-kernel/src/IHaskell/IPython/Message/Writer.hs]: uncomment or remove

File: ipython-kernel/src/IHaskell/IPython/Message/Writer.hs [hunk 5]
```diff
@@ -7,12 +7,12 @@ module IHaskell.IPython.Message.Writer (ToJSON(..)) where
 
 import           Data.Aeson
 import           Data.Map (Map)
+-- import           Data.Monoid (mempty)
```
Comment (gibiansky) [inline ipython-kernel/src/IHaskell/IPython/Message/Writer.hs]: uncomment or remove

**Comment by gibiansky** [comment]
Looks good, just a few very minor comments remaining. Thanks for doing this, this is really helpful, I've actually needed this myself recently but didn't get around to it.

What are you using this for (what sort of front-end are you making)?

**Comment by thomasjm** [comment]
Oh sorry, I didn't mean for you to review that commit yet. Emacs haskell-mode was being a little overzealous in commenting unused imports. I'll get a finalized commit out later today.

I'm making a frontend that lets you embed code in a Markdown file -- I'll post it publicly pretty soon

**Comment by thomasjm** [comment]
Sorry @gibiansky , this fell off my radar for a while. I just updated it and addressed your comments, could we get this merged?

**Pull request merged**